src/main/java/bio/terra/janitor/db/JanitorDao.java

```diff
@@ -70,20 +75,31 @@ public UUID createResource(
               .map(
                   entry ->
                       new MapSqlParameterSource()
-                          .addValue("tracked_resource_id", trackedResourceId)
+                          .addValue("tracked_resource_id", trackedResourceId.getUUID())
                           .addValue("key", entry.getKey())
                           .addValue("value", entry.getValue()))
               .toArray(size -> new MapSqlParameterSource[labels.size()]);
       try {
         jdbcTemplate.batchUpdate(insertLabelSql, sqlParameterSourceList);
       } catch (DuplicateKeyException e) {
-        throw new DuplicateLabelException(
+        throw new DuplicateDbKeyException(
             "Duplicate label found for tracked_resource_id: "
```

Comment:
I'm not sure this is worth catching and rethrowing specifically. But worrying about generating a conflicting UUID seems odd.
Thinking about this for the labels as well, with a Map<String, String> how can we have conflicting keys?

Reply:
Done. Removed this classes. I saw WM did that, so I did the same thing. But I now I am guessing the UUID is part of their request body, so it worth for them to catch it and throw a BadRequestException